pika/adapters/base_connection.py

```diff
@@ -382,20 +365,19 @@ def _handle_events(self, fd, events, error=None, write_only=False):
             self._handle_write()
             self._manage_event_state()
 
-        if self.socket and not write_only and (events & self.READ):
```

Comment:
`write_only` was most likely a vestige of the original `BlockingConnection` implementation prior to my rewrite of `BlockingConnection`.